pkg/registry/apps/deployment/storage/storage.go

```diff
@@ -401,8 +405,15 @@ func (i *scaleUpdatedObjectInfo) UpdatedObject(ctx context.Context, oldObj runti
 		)
 	}
 
-	// move replicas/resourceVersion fields to object and return
+	live := deployment.DeepCopy()
```

Comment:
I'm assuming that we need to copy in case the `Update` changes the object in place. I'm pretty sure it doesn't (do you remember @jpbetz?). Either way, there is no reason to copy if we don't call `Update`, at least move the copy in the if statement. But I don't think it's needed.

Reply:
I might be wrong, have you experimented with it?

Reply:
My assumption was that we need to pass to `Update` two objects: 1. the object "before" changing it and 2. the object "after" changing it so that the managed fields are updated accordingly. If we don't copy it we effectively pass the same object. I am probably missing something though. 

If we need to keep the copy, and we decide we want to do it only when necessary, the resulting code is a bit convoluted. I would propose something like:
```go
	var live *apps.Deployment
	if i.parentFieldManager != nil {
		live = deployment.DeepCopy()
	}

	// move replicas/resourceVersion fields to object
	deployment.Spec.Replicas = scale.Spec.Replicas
	deployment.ResourceVersion = scale.ResourceVersion

	if i.parentFieldManager != nil {
		// update managed fields using the fieldmanager of the main resource
		return i.parentFieldManager.Update(live, deployment, i.manager)
	}
	return deployment, nil
```
WDYT?